receiver/dotnetdiagnosticsreceiver/metrics/converter.go

```diff
@@ -0,0 +1,83 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package metrics
+
+import (
+	"math"
+	"time"
+
+	"go.opentelemetry.io/collector/consumer/pdata"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/dotnetdiagnosticsreceiver/dotnet"
+)
+
+func rawMetricsToPdata(rawMetrics []dotnet.Metric, now time.Time) pdata.Metrics {
+	pdm := pdata.NewMetrics()
+	rms := pdm.ResourceMetrics()
+	rms.Resize(1)
+	rm := rms.At(0)
+	rm.Resource().Attributes()
+	ilms := rm.InstrumentationLibraryMetrics()
+	ilms.Resize(1)
+	ilm := ilms.At(0)
+	ms := ilm.Metrics()
+	ms.Resize(len(rawMetrics))
+	for i := 0; i < len(rawMetrics); i++ {
+		rawMetricToPdata(rawMetrics[i], ms.At(i), now)
+	}
+	return pdm
+}
+
+func rawMetricToPdata(dm dotnet.Metric, pdm pdata.Metric, now time.Time) pdata.Metric {
+	const metricNamePrefix = "dotnet."
+	pdm.SetName(metricNamePrefix + dm.Name())
+	pdm.SetDescription(dm.DisplayName())
+	pdm.SetUnit(mapUnits(dm.DisplayUnits()))
+	nowPD := pdata.TimestampFromTime(now)
+	switch dm.CounterType() {
+	case "Mean":
+		pdm.SetDataType(pdata.MetricDataTypeDoubleGauge)
+		dps := pdm.DoubleGauge().DataPoints()
+		dps.Resize(1)
+		dp := dps.At(0)
+		dp.SetTimestamp(nowPD)
+		dp.SetValue(dm.Mean())
+	case "Sum":
+		pdm.SetDataType(pdata.MetricDataTypeDoubleSum)
+		sum := pdm.DoubleSum()
+		sum.SetAggregationTemporality(pdata.AggregationTemporalityDelta)
+		dps := sum.DataPoints()
+		dps.Resize(1)
+		dp := dps.At(0)
+		interval := time.Duration(math.Round(float64(dm.IntervalSec()) * float64(time.Second)))
```

Comment:
On second thought I'm going to rework this a little to use the previous timestamp instead of the reported interval duration.